spark/src/test/java/org/apache/iceberg/spark/actions/TestNewRewriteDataFilesAction.java

```diff
@@ -590,6 +595,156 @@ public void testInvalidOptions() {
             .execute());
   }
 
+  @Test
+  public void testSortMultipleGroups() {
+    Table table = createTable(20);
+    shouldHaveFiles(table, 20);
+    table.replaceSortOrder().asc("c2").commit();
+    int fileSize = averageFileSize(table);
+
+    List<Object[]> originalData = currentData();
+
+    // Perform a rewrite but only allow 2 files to be compacted at a time
+    RewriteDataFiles.Result result =
+        basicRewrite(table)
+            .sort()
+            .option(SortStrategy.REWRITE_ALL, "true")
+            .option(RewriteDataFiles.MAX_FILE_GROUP_SIZE_BYTES, Integer.toString(fileSize * 2 + 100))
+            .option(BinPackStrategy.MIN_INPUT_FILES, "1")
```

Comment:
Why use `BinPackStrategy.MIN_INPUT_FILES` with a sort strategy? Should this be moved or renamed?

Reply:
Looks like the next test uses `SortStrategy.MIN_INPUT_FILES`.

Reply:
I guess this isn't needed now that I added REWRITE_ALL, This can be removed now

Reply:
I forgot I set REWRITE ALL to only apply to filtering and not to the group limits, I should make it apply to both